### PR TITLE
Forward AutodIff Compatibility

### PR DESCRIPTION
**Pull request by JRock95**
I changed the internal time to be handled as seconds past JD_J2000 instead of DateTime to allow for automatic differentiation compatibility and changed some of the return types to handle more general inputs, namely ForwardDiff's Duals. Reverse Autodiff is blocked by not knowing how to handle the DCM struct at the moment but will be handled in a future PR.

The previous DateTime interfaces are still present they just internally convert the DateTime to the seconds past JD_J2000 and call the newer function.

### CODE REVIEW

File: src/GravityModels/accelerations.jl
Comment (ronisbr) [inline src/GravityModels/accelerations.jl]: If the parameter `W` is not used within the function, I think it is better to drop it and just use `time::Number`. AFAIR, this suggestion is written in BlueStyle or Julia documentation for coding guidelines.

File: src/GravityModels/accelerations.jl
Comment (ronisbr) [inline src/GravityModels/accelerations.jl]: Will it have an impact if we try to keep all the constants as `Integer` if possible? For example, use `86400` instead of `86400.0` and `-43200` instead of `-43200.0`. AFAIR, this approach is also mentioned somewhere in Julia documentation.

File: src/GravityModels/accelerations.jl
Comment (ronisbr) [inline src/GravityModels/accelerations.jl]: Same comment for unnecessary type parameters.

**Comment by ronisbr** [comment]
Hi @jmurphy6895 !

Aweome PR, thank you very much! I just added some general comments, check, please, if they apply.

**Comment by ronisbr** [comment]
Hi @jmurphy6895,

Can you please check if the failures are only related to test packages that cannot be installed in Julia 1.6? If so, can you modify the test script to ignore those tests in Julia 1.6? I plan to drop support for Julia 1.6 but it will take sometime to update all packages in SatelliteToolbox.jl ecosystem.

**Comment by JRock95** [comment]
Hey @ronisbr,

I don't see your comments on the PR, are they on a different branch or maybe not pushed?

As for the failure, yeah it looks like it's because of Julia 1.6:
`LoadError: Unsatisfiable requirements detected for package FiniteDiff`
...
`restricted by julia compatibility requirements to versions: 2.0.0-2.22.0 or uninstalled — no versions left`

I added a conditional to not run on v1.6

**Review by ronisbr — CHANGES_REQUESTED** [review]

**Comment by ronisbr** [comment]
Thanks @jmurphy6895 ! I forgot to push the comments :)

**Comment by JRock95** [comment]
I see why it's still failing, it's not the test it's the .toml compat, I'll loosen it and see if the tests still pass

**Comment by JRock95** [comment]
I'll get to these changes after work today

File: src/GravityModels/accelerations.jl
Comment (JRock95) [reply src/GravityModels/accelerations.jl]: Sounds good, removed

File: src/GravityModels/accelerations.jl
Comment (JRock95) [reply src/GravityModels/accelerations.jl]: Doesn't look like it I've changed this to ints around the PR

File: src/GravityModels/accelerations.jl
Comment (JRock95) [reply src/GravityModels/accelerations.jl]: Removed

**Comment by JRock95** [comment]
W.r.t. to the failing CI, it looks like the DifferentiationInterface will not work with Julia 1.6 for FiniteDIff (Julia 1.6 requires <=v2.22 and DifferentiationInterface requires >=2.23) 
Screenshot 2024-09-25 221841

I don't explicitly need the DifferentiationInterface package to test this, it will just make testing all the different AD backends very easy in the future. I'll defer to you on what you'd like to happen here

**Comment by ronisbr** [comment]
Hi @jmurphy6895 !

Let's ignore those errors. As soon as we merge this commit, I turn off the tests for Julia 1.6. The last thing (sorry, I missed it before): can you please return the version to 0.1.5? I prefer to have a separate commit with the version bump. After that, I will merge this PR and tag the new release :)

**Comment by JRock95** [comment]
@ronisbr Done!

**Comment by ronisbr** [comment]
Awesome! Let's the CI run and I will merge this! Thank you very much for the PR @jmurphy6895 !

**Comment by ronisbr** [comment]
Hi @jmurphy6895 !

I did some modifications:

1. Fixed precompilation warnings.
2. Improved docstrings (making them less verbose).
3. Most important: fixed J2000.0 epoch definition. Your PR revealed a long-lasting bug between the code and the documentation. When we pass no time information, we fallback to J2000.0 epoch, which is 2000-01-01T12:00:00 instead of 2000-01-01T00:00:00. Hence, I modified everything to provide the coefficients at the correct definition of J2000.0 epoch.

**Comment by ronisbr** [comment]
Can you please test if I broke something related to autodiff? If not, I will tag the new release.

**Comment by JRock95** [comment]
LGTM, one step closer to a fully differentiable propagator!

**Pull request merged**
